### ISSUE TITLE
[enhancement] Custom editor "textbox" for editing large text in textarea

### ISSUE THREAD

**Issue by chaissac** [olifolkerd/tabulator#45]
Bonjour, hi,

I had a large text (comment, etc.) to edit in a table.
I used a textarea and create a custom editor "textbox" for this.
I think it could be useful for many people, so I share it below : 

```
		textbox:function(cell, value){
			//create and style input
			var input = $("<textarea></textarea>");
			input.css({
				"border":"1px",
				"display":"block",

				"position":"absolute",
				"background":"#fff",
				"padding":"2px",
				"width":cell.width()+"px",
				"height":(cell.height()*8)+"px",
				"box-sizing":"border-box",
				"white-space":"pre-wrap",
				"overflow":"visible"
			})
			.val(value);

			setTimeout(function(){
				input.focus();
			},100);

			//submit new value on blur
			input.on("change blur", function(e){
				cell.trigger("editval", input.val());
			});

			return input;
		},

```

Thx !
Philippe

**Comment by olifolkerd** [comment]
That is a great idea! i will add something in this week.

Cheers
Oli

**Comment by chaissac** [comment]
The trick here is to set an "absolute" position to the textarea ; I first try to modify the height of all the cells of the row, but it was pretty dirty and required to store the height, and to add some code elsewhere to resize the row. The CSS solution is more elegant. There's only one little problem : the textarea is not restored to its original size when you try to move a row while editing... But it's not really inconvenient.

The important css properties are : "display":"block", "position":"absolute", "width":cell.width()+"px", "height":(cell.height()*8)+"px", "white-space":"pre-wrap", "overflow":"visible". The 8 number for the height should be changed with an option or a parameter - but for my usage, 8 lines is perfect. A css class could be added for this textarea in the tabulator.css.

Hope it helps :)
Cheers
Philippe

**Comment by olifolkerd** [comment]
I have just released version 2.4 which has full multi-line text support.

A new ***textarea*** formatter and editor have been added to allow for the display and editing of multi-line text.

The ***textarea*** formatter allows the row to expand vertically to accommodate multiple lines of text. 
Check out the [Formatter Documentation](http://olifolkerd.github.io/tabulator/docs/#formatting).

The ***textarea*** editor allwos you to edit multi-line text, and will expand and contract vertically as the the number of lines change.
Check out the [Editor Documentation](http://olifolkerd.github.io/tabulator/docs/#manipulating-data).
